spec/support/pageflow/entries_controller_test_helper.rb

```diff
@@ -0,0 +1,50 @@
+module Pageflow
+  # Helpers to test entry type controllers.
+  #
+  # @since edge
+  module EntriesControllerTestHelper
+    extend ActiveSupport::Concern
+
+    included do
+      # Entry type controllers are delegated to by
+      # `Pagefow::EntriesController`. Therefore they do not have their
+      # own routes. Controller specs only allow calling actions that
+      # have a route. So we create a fake route to work aroung this.
+
+      controller_name = described_class.name.sub(/Controller$/, '').underscore
+
+      routes do
+        ActionDispatch::Routing::RouteSet.new.tap do |routes|
+          routes.draw do
+            get '/test/entry', to: "#{controller_name}#show"
+          end
+        end
+      end
+    end
+
+    # Invoke an action of the entries controller with the request env
+    # set up just like when Pageflow delegates to the entry type's
+    # frontend app.
+    #
+    # @param action [Symbol] Name of the action to invoke
+    #
+    # @param entry [Entry] Entry to render.
+    #
+    # @param mode [:published|:preview] Whether to render the
+    #   published revision or the draft.
+    def get_with_entry_env(action, entry:, mode: :published, params: {})
+      revision =
+        if mode == :published
+          entry.published_revision
+        else
+          entry.draft
+        end
+
+      published_entry = PublishedEntry.new(entry, revision)
```

Comment:
How come we're always using `PublishedEntry` now, even when it's just a `draft` (i.e. `mode: :preview`)?

Reply:
That's also [the way it used to work before](https://github.com/codevise/pageflow/blob/master/app/controllers/pageflow/revisions_controller.rb#L13). `DraftEntry` has additional methods for editing entries (adding files, updating meta data). Whenever we are viewing an entry outside of the editor (whether it's a preview or the real thing), `PublishedEntry` is used.